source/common/network/addr_family_aware_socket_option_impl.cc

```diff
@@ -29,7 +30,7 @@ absl::optional<Address::IpVersion> getVersionFromSocket(const Socket& socket) {
     if (socket.localAddress()) {
       return {getVersionFromAddress(socket.localAddress())};
     } else {
-      return {getVersionFromAddress(Address::addressFromFd(socket.ioHandle().fd()))};
+      return {getVersionFromAddress(SocketInterface::fdLocalAddress(socket.ioHandle().fd()))};
```

Comment:
Seems like we should avoid the fd back and forth here and just have a method directly on the socket or iohandle?

Reply:
Agreed! I didn't change this because I wanted to avoid surprises when the socket was not properly initialized (still learning the code :)). 

I guess we could store the version in the socket (together with the type) at which point this whole function becomes redundant. It would be nice not to have to, but one could allocate a socket without assigning any local/remote address. Unless, we explicitly disallow that.